core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -220,6 +222,15 @@ class ZooKeeperClient(connectString: String, sessionTimeoutMs: Int, connectionTi
     zNodeChildChangeHandlers.remove(path)
   }
 
+  def registerStateChangeHandler(statusChangeHandler: StateChangeHandler): Unit = {
```

Comment:
With this, do we still need stateChangeHandler in the constructor?

Reply:
@junrao  There will a small time gap between ZooKeeperClient start and call to registerStateChageHanlder method. So there is a chance of missing  initial state events.
So i felt, we can retain stateChangeHandler in constructor and we can also allow to register  to using registerStateChageHanlder method.

Components like DynamicConfigManager, ZKNotif which are registering after ZooKeeperClient initialization , can register using registerStateChageHanlder and handle state changes.

cc  @onurkaraman 
Updated the PR with other review comments.